### PR TITLE
Add functionality like move_ee_xyz, but for rotation about a single EE axis at a time.

### PR DESCRIPTION
**Pull request by anthonysimeonov**
Added rot_ee_xyz to single_arm_pybullet.py, which allows for similar functionality to move_ee_xyz but with rotations about the end effector axes instead of just translations. Along with this is a new utils file, transform_util.py, that has some functionality for manipulating 6D poses. Added rot_ee_xyz example to ur/sim examples.

### CODE REVIEW

File: src/airobot/arm/single_arm_pybullet.py [hunk 1]
```diff
@@ -320,6 +321,52 @@ def move_ee_xyz(self, delta_xyz, eef_step=0.005, *args, **kwargs):
...
+            bool: A boolean variable representing if the action is successful
+            at the moment when the function exits.            
+        """
+        if not self._pb.in_realtime_mode():
+            raise AssertionError('rot_ee_xyz() can  '
+                                 'only be called in realtime'
+                                 ' simulation mode')
+
+        axis_dict = {'x': 0, 'y': 1, 'z': 2}
+        if axis not in ['x', 'y', 'z']:
```
Comment (taochenshh) [inline src/airobot/arm/single_arm_pybullet.py:346]: maybe better to use `if axis not in axis_dict` to remove duplicate code?

File: src/airobot/utils/transform_util.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        np.ndarray: Homogeneous transformation matrix representation of 
+            the pose.
+    """
+    pose_list = pose_stamped2list(pose)
+    trans, quat = pose_list[:3], pose_list[3:]
+    T = np.eye(4)
+    if hasattr(R, 'as_matrix'):
+        T[:-1, :-1] = R.from_quat(quat).as_matrix()
+    else:
+        T[:-1, :-1] = R.from_quat(quat).as_dcm()
```
Comment (taochenshh) [inline src/airobot/utils/transform_util.py:157]: can we replace these lines with the existing functions in common.py

File: src/airobot/utils/transform_util.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+            shape :math:`4x4`.
+        frame_id (str, optional): Reference frame of the pose. 
+
+    Returns:
+        PoseStamped: 6D pose representation of the transformation matrix.
+    """
+    if hasattr(R, 'from_matrix'):
+        quat = R.from_matrix(matrix[:-1, :-1]).as_quat()
+    else:
+        quat = R.from_dcm(matrix[:-1, :-1]).as_quat()
```
Comment (taochenshh) [inline src/airobot/utils/transform_util.py:178]: same for these maybe

File: src/airobot/utils/transform_util.py [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+    interp_rots = slerp(np.linspace(0, 1, N))
+    quat_interp_total = interp_rots.as_quat()    
+
+    pose_interp = []
+    for counter in range(N):
+        pose_tmp = [
+            trans_interp_total[0][counter],
+            trans_interp_total[1][counter],
+            trans_interp_total[2][counter],
+            quat_interp_total[counter][0], #return in ROS ordering w,x,y,z
```
Comment (taochenshh) [inline src/airobot/utils/transform_util.py:320]: is the comment here outdated? seems like the ordering is x,y,z,w

**Review by taochenshh — CHANGES_REQUESTED** [review]

File: src/airobot/utils/transform_util.py (hunk 4 above)
Comment (anthonysimeonov) [reply src/airobot/utils/transform_util.py]: fixed in 2259946

File: src/airobot/utils/transform_util.py (hunk 3 above)
Comment (anthonysimeonov) [reply src/airobot/utils/transform_util.py]: fixed in 2259946

File: src/airobot/utils/transform_util.py (hunk 2 above)
Comment (anthonysimeonov) [reply src/airobot/utils/transform_util.py]: fixed in 2259946

File: src/airobot/arm/single_arm_pybullet.py (hunk 1 above)
Comment (anthonysimeonov) [reply src/airobot/arm/single_arm_pybullet.py]: fixed in 2259946

**Review by taochenshh — APPROVED** [review]

**Pull request merged**